MdeModulePkg/Core/Dxe/Image/Image.c

```diff
@@ -1768,7 +1768,7 @@ CoreStartImage (
   // Image has completed.  Verify the tpl is the same
   //
   ASSERT (Image->Tpl == gEfiCurrentTpl);
-  CoreRestoreTpl (Image->Tpl);
+  // CoreRestoreTpl (Image->Tpl); //MU_CHANGE - remove unecessary TPL restore.

```

Comment:
@joschock do you think we could be a bit more conservative here and conditionalize the change so prior behavior can be maintained if needed?

```suggestion
  if (Image->Tpl != gEfiCurrentTpl) { // MU_CHANGE - reduce TPL restore
    CoreRestoreTpl (Image->Tpl); 
  } // MU_CHANGE - reduce TPL restore
```

Reply:
Sure, I can make that change.